### PR TITLE
Prevent duplicate channel instantiation and joins

### LINKED ISSUE #3171: phoenix.js should close channel when there is a duplicate join without waiting for the server
Otherwise we can have race conditions in the client as two channels are running at the same time.

### PR DESCRIPTION
**Pull request by dballona**
Fixes #3171

### CODE REVIEW

File: assets/test/channel_test.js [hunk 1]
```diff
@@ -105,6 +105,17 @@ describe("join", () => {
     assert.throws(() => channel.join(), /^Error: tried to join multiple times/)
   })
 
+  it("prevents duplicate join when race condition", () => {
+    channel.join()
```
Comment (achempion) [inline assets/test/channel_test.js]: shall we check for `assert.equal(socket.channels.length, 0)`?

Comment (dballona) [reply assets/test/channel_test.js]: sure! I'll add a separate test case for this assertion.

Comment (dballona) [reply assets/test/channel_test.js]: I just realized this spec is actually not impacting this implementation at all as a `channel` is only pushed to the socket `channels` on initialization... `channel.join` has zero impact on that : )

Also, I realized we're not handling the case for duplicate join properly – working on it!

**Comment by dballona** [comment]
I just realized the multiple join implementation I did doesn't make any sense. 
Will close the PR again until I get a reasonable solution... sorry about this.

**Pull request closed without merging**